quickwit-search/src/search_response_rest.rs

```diff
@@ -48,12 +49,26 @@ impl TryFrom<quickwit_proto::SearchResponse> for SearchResponseRest {
             .hits
             .into_iter()
             .map(|hit| {
-                serde_json::from_str(&hit.json).map_err(|err| {
-                    SearchError::InternalError(format!(
-                        "Failed to serialize document `{}` to JSON: `{}`.",
-                        hit.json, err
-                    ))
-                })
+                let mut hit_value = Map::with_capacity(2);
+                let document: serde_json::Value =
+                    serde_json::from_str(&hit.json).map_err(|err| {
+                        SearchError::InternalError(format!(
+                            "Failed to serialize document `{}` to JSON: `{}`.",
```

Comment:
I don't think displaying the whole doc is a good idea. Maybe the first n chars?

Reply:
agreed... Be careful there. @guilload mentioned chars because you cannot truncate bytes as it might end up being an invalid string. `.chars().take().collect()`, or if you want to avoid the alloc, write a `truncate_str` helper function relying on `str.is_char_boundary()` and add it to quickwit-common.